src/wendy/commands.clj

```diff
@@ -81,6 +81,15 @@
                     number)]
     (effects/request url :post)))
 
+(defn pipeline-list!
+  [group name status]
+  (let [url (format "%s/pipelines?group=%s&name=%s&status=%s"
+                    (bob-url)
+                   (str group)
```

Comment:
This won't work if the options are not passed. They would be nil and we need to
check if name, group and stauts are nil and they should not be in the query
params.

Reply:
But `(str nil)` will result in an empty string. I thought this will be less complicated code than optionally adding th params. What do you think?

Reply:
yeah but it doesn't look clean and could be error prone. can try something like this in a function which takes a map of options like `{:k1 "v1" :k2 nil :k3 "v3"}` and gives back `"?k1=v1&k3=v3"`.

```clojure
(->> {:k1 "v1" :k2 nil :k3 "v3"}
     (filter #(some? (second %)))
     (map #(str (name (first %)) "=" (str (second %))))
     (clojure.string/join "&")
     (#(if (not-empty %) (str "?" %) "")))
```

what do you think? this is just something i hacked together and could be written better.